### ISSUE TITLE
Feature Request: JavaScript Implementation

### ISSUE THREAD

**Issue by aboy021** [terrastruct/d2-obsidian#19]
I realise that this is a big ask, but the current plugin is really just a wrapper for a command line application. It would be great if it was a JS plugin that worked out of the box and worked on mobile devices too. I think there would be broader benefits to D2 as well.

**Comment by alixander** [comment]
yup, D2 is all in Go, which is compilable to WebAssembly, which can run in JS contexts. this is in progress: https://github.com/terrastruct/d2/pull/436

**Comment by hpr1999** [comment]
I made a prototype [here](https://github.com/terrastruct/d2-obsidian/pull/56) if anyone would like to discuss.
But considering it doesn't seem to have TALA support yet, maybe it's just not quite ready? =)
